site/content/protocols/messagepickup/4.0/readme.md

```diff
@@ -0,0 +1,389 @@
+
+The `live-delivery-change` message is used to set the state of `live_delivery`.
+
+When _Live Mode_ is enabled, messages that arrive when an existing connection exists are delivered over the connection immediately, via a `delivery` message, rather than being pushed to the queue. See _Live Mode_ below for more details. 
+
+## Security
+
+This protocol expects messages to be encrypted during transmission, and repudiable. 
+
+## Message Reference
+
+### Status Request
+Sent by the `recipient` to the `mediator` to request a `status` message.
+
+Message Type URI: `https://didcomm.org/message-pickup/4.0/status-request`
+
+DIDComm v1 example:
+```json
+{
+    "@id": "123456780",
+    "@type": "https://didcomm.org/message-pickup/4.0/status-request",
+    "recipient_did": "<did:key for messages>",
+    "~transport": {
+        "return_route": "all"
+    }
+}
+```
+
+DIDComm v2 example:
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/message-pickup/4.0/status-request",
+    "body" : {
+        "recipient_did": "<did for messages>"
+    },
+    "return_route": "all"
+}
+```
+`recipient_did` is optional. When specified, the `mediator` **MUST** only return status related to that recipient did. This allows the `recipient` to discover if any messages are in the queue that were sent to a specific DID. If using DIDComm v1, `recipient_did` **MUST** be a [`did:key` reference](https://github.com/hyperledger/aries-rfcs/tree/main/features/0360-use-did-key).
+
+### Status
+Status details about waiting messages.
+
+Message Type URI: `https://didcomm.org/message-pickup/4.0/status`
+
+DIDComm v1 example:
+```json
+{
+    "@id": "123456780",
+    "@type": "https://didcomm.org/message-pickup/4.0/status",
+    "~thread": {
+        "thid": "<message id of status-request message>"
+    },
+    "recipient_did": "<did:key for messages>",
+    "message_count": 7,
+    "longest_waited_seconds": 3600,
+    "newest_received_time": 1658085169,
+    "oldest_received_time": 1658084293,
+    "total_bytes": 8096,
+    "live_delivery": false
+}
+```
+
+DIDComm v2 example:
+```json
+{
+    "id": "123456780",
+    "thid": "<message id of status-request message>",
+    "type": "https://didcomm.org/message-pickup/4.0/status",
+    "body": {
+        "recipient_did": "<did for messages>",
+        "message_count": 7,
+        "longest_waited_seconds": 3600,
+        "newest_received_time": 1658085169,
+        "oldest_received_time": 1658084293,
+        "total_bytes": 8096,
+        "live_delivery": false
+    }
+}
+```
+
+`message_count` is the only **REQUIRED** attribute. The others **MAY** be present if offered by the `mediator`.
+
+`longest_waited_seconds` is the age, in seconds, of the oldest message in the queue.
+
+`newest_received_time` and `oldest_received_time` are expressed in UTC Epoch Seconds (seconds since 1970-01-01T00:00:00Z) as an integer.
```

Comment:
We discussed this in the DIDComm UG on 12-9-24--we determined that we wanted to discuss with additional folks in the community to get their takes on a general best practice for DIDComm and also bring it for discussion in the WG spec meeting. 
Specifically for this protocol we have determined we want additional specificity, but the debate is whether to do ISO 8601 or do UTC Epoch Milliseconds. I now lean towards the UTC approach since it's represented as an integer instead of a string.